control_plane/scripts/uninstall.py

```diff
@@ -370,11 +380,18 @@ def resource_group_summary(resource_groups: list[str]) -> str:
 
 
 def uninstall_summary(
-    sub_to_rg_deletions: dict[str, list[str]],
     sub_id_to_name: dict[str, str],
-    sub_to_role_assignment_deletions: dict[str, list[dict[str, str]]],
-    sub_to_resource_id_to_diag_setting_deletions: dict[str, dict[str, list[str]]],
+    sub_to_rg_deletions: SubIdToResourceGroupsDict,
+    sub_to_role_assignment_deletions: SubIdToRoleAssignmentsDict,
+    sub_to_resource_id_to_diag_setting_deletions: SubIdToResourceIdToDiagSettingsDict,
 ) -> str:
+    if (
+        not sub_to_rg_deletions
```

Comment:
all the nots make this harder to reason about but not sure how convinced I am that this is much clearer 
```python
if not (
    sub_to_rg_deletions ||
    sub_to_role_assignment_deletions ||
    sub_to_resource_id_to_diag_setting_deletions
)
```